azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -0,0 +1,132 @@
+ * be added here as a failsafe if needed.
+ */
+@Singleton
+public class ContainerCleanupManager {
+
+  private static final Logger logger = LoggerFactory.getLogger(ContainerCleanupManager.class);
+  private static final Duration DEFAULT_STALE_CONTAINER_CLEANUP_INTERVAL = Duration.ofMinutes(10);
+  private static final Duration DEFAULT_STALE_CONTAINER_AGE = Duration.ofDays(10);
+  private final long cleanupIntervalMin;
+  private final long staleContainerAgeDays;
+  private final ScheduledExecutorService cleanupService;
+  private final ExecutorLoader executorLoader;
+  private final ContainerizedImpl containerizedImpl;
+
+  @Inject
+  public ContainerCleanupManager(final Props azkProps, final ExecutorLoader executorLoader,
+      final ContainerizedImpl containerizedImpl) {
+    this.cleanupIntervalMin = azkProps
+        .getLong(
+            ContainerizedExecutionManagerProperties.CONTAINERIZED_STALE_EXECUTION_CLEANUP_INTERVAL_MIN,
+            DEFAULT_STALE_CONTAINER_CLEANUP_INTERVAL.toMinutes());
+    this.staleContainerAgeDays = azkProps
+        .getLong(
+            ContainerizedExecutionManagerProperties.CONTAINERIZED_STALE_EXECUTION_AGE_DAYS,
+            DEFAULT_STALE_CONTAINER_AGE.toDays());
+    this.cleanupService = Executors.newSingleThreadScheduledExecutor(
+        new ThreadFactoryBuilder().setNameFormat("azk-container-cleanup").build());
+    this.executorLoader = executorLoader;
+    this.containerizedImpl = containerizedImpl;
+  }
+
+  /**
+   * Execute container-provider specific APIs for all 'stale' containers. A container is considered
+   * 'stale' if it was launched CONTAINERIZED_STALE_EXECUTION_AGE_DAYS ago and the corresponding
+   * execution is not yet in a final state.
+   * <p>
+   * It's important that this method does not throw exceptions as that will interrupt the scheduling
+   * of {@code cleanupService}.
+   */
+  public void terminateStaleContainers() {
+    try {
+      final List<ExecutableFlow> staleFlows = this.executorLoader.fetchStaleFlows(Duration.ofDays(
+          this.staleContainerAgeDays));
+      for (final ExecutableFlow flow : staleFlows) {
+        deleteContainerQuietly(flow.getExecutionId());
+      }
+    } catch (final Exception e) {
+      logger.error("Unexpected exception during container cleanup." + e);
+    }
+  }
+
+  // Deletes the container specified by executionId while logging and consuming any exceptions.
+  // Note that while this method is not async it's still expected to return 'quickly'. This is true
+  // for Kubernetes as it's declarative API will only submit the request for deleting container
+  // resources. In future we can consider making this async to eliminate any chance of the cleanup
+  // thread getting blocked.
+  private void deleteContainerQuietly(final int executionId) {
+    try {
+      this.containerizedImpl.deleteContainer(executionId);
+    } catch (final ExecutorManagerException eme) {
```

Comment:
If possible, you should catch ApiException and use e.getResponseBody() for detail message

Reply:
`ApiException` will be specific to the k8s Containerized implementation. This layer is interacting through the interface `ContainerizedImpl` (similar to a previous comment)

Reply:
Point noted. I already have added logs for error this case. It will be taken care of.